core/lib/utils/image_display/image_display.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'image_preview.dart';
+
+class AppImageDisplay {
+  static final AppImageDisplay _instance = AppImageDisplay._();
+
+  AppImageDisplay._();
+
+  factory AppImageDisplay() => _instance;
+
+  Padding buildImages({
```

Comment:
I would anticipate this not merely as a function, but rather as a versatile widget that accepts parameters. We can call it `ImageGallery` or something similar